### DISCUSSION TITLE
how to declare type for defaultdict(defaultdict)

### DISCUSSION THREAD

**Discussion by matt-the-midnight-hacker** [microsoft/pyright#8839] [Q&A] [answered]
Hi I just updated pyright to 1.1.377 and I get now type errors for `defaultdict(defaultdict)` assignments. I'm not sure if it is a bug or a typeshed issue or just me.

The code to reproduce what I mean would be:
```
from typing import DefaultDict
from collections import defaultdict

class MyClass:
    pass

d: DefaultDict[int, DefaultDict[int, MyClass]] = defaultdict(defaultdict)

d[1][2] = MyClass()
```
I get error:

`error: Expression of type "defaultdict[Unknown, defaultdict[Unknown, defaultdict[Unknown, Unknown]]]" is incompatible with declared type "DefaultDict[int, DefaultDict[int, MyClass]]`

How to do the `defaultdict` instantiation here to fulfill the type I've declared?

fyi. I've also asked this on [stackoverflow](https://stackoverflow.com/questions/78917316/how-to-instantiate-defaultdict-to-fulfill-defaultdictint-defaultdictint-mycl)

**Comment by erictraut** [comment, marked as answer]
On first inspection, this looks like a bug. I'll create an issue from the discussion. In the meantime, you can work around the issue this way:

```python
d: DefaultDict[int, DefaultDict[int, MyClass]] = defaultdict(defaultdict[int, MyClass])
```

This workaround might not work on older versions of Python where `defaultdict` was not indexable.

**Reply by erictraut** [reply]
Here's the issue: https://github.com/microsoft/pyright/issues/8840.

**Reply by matt-the-midnight-hacker** [reply]
thank you so much

**Reply by tofetpuzo** [reply]
from my stackoverflow answer, please as this issue be resolved?
